packages/testing-core/src/lib/expectToBeDefined.ts

```diff
@@ -0,0 +1,10 @@
+import { ok } from "node:assert";
+
+/**
+ * Asserts and narrows the type for defined values.
+ *
+ * @throws {AssertionError} for null or undefined values.
+ */
+export function expectToBeDefined<T>(value: T | undefined): asserts value is T {
```

Comment:
how is this different than https://jestjs.io/docs/expect#tobedefined ?

Reply:
It type narrows, the Jest version does not.

Given:

```typescript
function getValue(): string | undefined {
  return "hi";
}
```

This gives a type error:

```typescript
const value = getValue();

expect(value).toBeDefined();

const { length } = value;
// ^? Property 'length' does not exist on type 'string | undefined'.
```

But this properly narrows the type:

```typescript
const value = getValue();

expectToBeDefined(value);

// Narrowed to `string`
const { length } = value;
expect(length).toBe(4);
```